### PR TITLE
NAS-130116 / 24.10 / Fix ipa join admin privileges and keytab

### PR DESCRIPTION
**Pull request by anodos325**
This adds back in missing cherry-pick from development branch to properly grant privileges to the admins group in an IPA domain.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/ldap.py
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/ldap.py:928]: The `set_state` to `HEALTHY` here doesn't seem correct.  We _could_ have just set it to JOINING above.

_Maybe_ indent under
```
if dom_join_resp in (DomainJoinResponse.ALREADY_JOINED.value, DomainJoinResponse.PERFORMED_JOIN.value):
```
line?

(Actually now I'm wondering whether we need this same condition above too.)

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/ldap.py
Comment (anodos325) [reply src/middlewared/middlewared/plugins/ldap.py:900]: Derp. Yup.

File: src/middlewared/middlewared/plugins/ldap.py
Comment (anodos325) [reply src/middlewared/middlewared/plugins/ldap.py:928]: I'll add comments.

**Review by mgrimesix — APPROVED** [review]
My changes were corrected, might wait for Brian's approval.

File: tests/api2/test_ipa_join.py
Comment (bmeagherix) [inline tests/api2/test_ipa_join.py:88]: priv is now undefined below

Also 2 flake8 indentation is not a multiple of 4 on lines 21 & 30 above

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/ldap.py
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/ldap.py:945]: Typo: `dom_join_response` should be `dom_join_resp`

**Review by bmeagherix — APPROVED** [review]
Approve, but one fix still required.

**Pull request merged**
